### PR TITLE
Fix right click camera pan behavior

### PR DESCRIPTION
**Pull request by charliefoxtwo**
This should bring the right-click camera pan system more closely in line with Generals/ZH.

### CODE REVIEW

**Comment by Tarcontar** [comment]
This is looking good for me. I have no idea how this is affected by the multiple? control options in generals (we recently switched to the old version).

**Comment by charliefoxtwo** [comment]
Are you referring to the alternate click mode (right click to issue move orders, etc)? I was thinking about that one yesterday... Will require more thought to implement though. Or are you referring to something else?

File: src/OpenSage.Game/Logic/SelectionSystem.cs [hunk 1]
```diff
@@ -23,20 +24,32 @@ public enum SelectionStatus
...
+
         private SelectionGui SelectionGui => Game.Scene3D.SelectionGui;
 
-        private Point2D _startPoint;
-        private Point2D _endPoint;
+        private Point2D _selectionStartPoint;
+        private Point2D _selectionEndPoint;
+
+        private Point2D _panStartPoint;
+        private readonly Stopwatch _panStopwatch = new Stopwatch();
```
Comment (charliefoxtwo) [inline src/OpenSage.Game/Logic/SelectionSystem.cs:41]: Looking through more of the solution, I'm actually not convinced using a stopwatch is the best course of action here. Should I be using `TimeInterval` and `Game.MapTime` instead? That seems to be the established pattern elsewhere.

Comment (Tarcontar) [reply src/OpenSage.Game/Logic/SelectionSystem.cs:41]: Indeed stopwatch is not the preferred way to do it.

**Comment by Tarcontar** [comment]
> Are you referring to the alternate click mode (right click to issue move orders, etc)? I was thinking about that one yesterday... Will require more thought to implement though. Or are you referring to something else?

Yes that was what i was referring to, i am not an expert tho. @tgjones might know more about that.

File: src/OpenSage.Game/Logic/SelectionSystem.cs (hunk 1 above)
Comment (feliwir) [reply src/OpenSage.Game/Logic/SelectionSystem.cs:41]: Yes, we don't use StopWatches anywhere usually

Comment (charliefoxtwo) [reply src/OpenSage.Game/Logic/SelectionSystem.cs:41]: I'll get that changed, thanks

**Comment by charliefoxtwo** [comment]
CI build failure seems to be unrelated?

Let me know if the updated timing code is any better. I mimicked what I saw in `GameObject.Construct()`, so hopefully that's the right idea.

**Comment by feliwir** [comment]
> CI build failure seems to be unrelated?

CI failure was unrelated, sorry about that

> Let me know if the updated timing code is any better. I mimicked what I saw in `GameObject.Construct()`, so hopefully that's the right idea.

Yep, that looks good now. I think we can merge this if @Tarcontar  approves

**Review by Tarcontar — APPROVED** [review]

**Comment by Tarcontar** [comment]
Looks good to me, thx!

**Pull request merged**